dynawo/sources/Modeler/DataInterface/IIDM/DYNDataInterfaceIIDM.cpp

```diff
@@ -1167,25 +1181,34 @@ DataInterfaceIIDM::configureBusCriteria(const boost::shared_ptr<criteria::Criter
       for (criteria::Criteria::component_id_const_iterator cmpIt = crit->begin(),
           cmpItEnd = crit->end();
           cmpIt != cmpItEnd; ++cmpIt) {
-        std::map<std::string, boost::shared_ptr<ComponentInterface> >::const_iterator busItfIter = components_.find(*cmpIt);
-        if (busItfIter != components_.end()) {
-          const boost::shared_ptr<ComponentInterface>& cmp = busItfIter->second;
-          if (cmp->getType() != ComponentInterface::BUS)
-            Trace::warn() << DYNLog(WrongComponentType, *cmpIt, "bus") << Trace::endline;
-          if (crit->hasCountryFilter()) {
-            boost::shared_ptr<BusInterfaceIIDM> bus = dynamic_pointer_cast<BusInterfaceIIDM>(cmp);
-            if (bus && !bus->getCountry().empty() && !crit->containsCountry(bus->getCountry()))
-              continue;
-            boost::shared_ptr<CalculatedBusInterfaceIIDM> calcBus = dynamic_pointer_cast<CalculatedBusInterfaceIIDM>(cmp);
-            if (calcBus && !calcBus->getCountry().empty() && !crit->containsCountry(calcBus->getCountry()))
+        boost::shared_ptr<BusInterface> bus;
+        if (!cmpIt->getVoltageLevelId().empty()) {
+          bus = findCalculatedBusInterface(cmpIt->getVoltageLevelId(), cmpIt->getId());
```

Comment:
It is not symetrical here: if a calculated bus is not found an error is thrown, if a bus is not found only a warning is displayed

Reply:
After discussion, conclusion is Warning for both.
Thus, `findCalculatedBusInterface` is reworked as mentioned above.